tests/main/mount-ns/google.ubuntu-16.04-64/PER-SNAP-18.expected.txt

```diff
@@ -54,21 +54,75 @@
 1054 1001 0:0 /tmp /tmp rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
 1055 1054 0:0 /tmp/snap.test-snapd-mountinfo-core18/tmp /tmp rw,relatime - ext4 /dev/sda1 rw,data=ordered
 1056 1001 2:0 /usr/lib/snapd /usr/lib/snapd ro,nodev,relatime master:16 - squashfs /dev/loop0 ro
-1057 1001 0:0 /usr/src /usr/src rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
-1058 1001 0:0 /var/lib/snapd /var/lib/snapd rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
-1060 1059 0:0 / /var/lib/snapd/hostfs rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
-1059 1058 0:0 /var/lib/snapd/hostfs /var/lib/snapd/hostfs rw,relatime - ext4 /dev/sda1 rw,data=ordered
-1061 1060 1:9 / /var/lib/snapd/hostfs/run rw,nosuid,noexec,relatime master:11 - tmpfs tmpfs rw,size=VARIABLE,mode=755
-1062 1061 1:10 / /var/lib/snapd/hostfs/run/cgmanager/fs rw,relatime master:12 - tmpfs cgmfs rw,size=VARIABLE,mode=755
-1063 1061 1:11 / /var/lib/snapd/hostfs/run/lock rw,nosuid,nodev,noexec,relatime master:13 - tmpfs tmpfs rw,size=VARIABLE
-1064 1061 1:12 / /var/lib/snapd/hostfs/run/rpc_pipefs rw,relatime master:14 - rpc_pipefs sunrpc rw
-1065 1061 1:9 /snapd/ns /var/lib/snapd/hostfs/run/snapd/ns rw,nosuid,noexec,relatime - tmpfs tmpfs rw,size=VARIABLE,mode=755
-1066 1061 1:13 / /var/lib/snapd/hostfs/run/user/0 rw,nosuid,nodev,relatime master:15 - tmpfs tmpfs rw,size=VARIABLE,mode=700
-1067 1060 2:0 / /var/lib/snapd/hostfs/snap/core/1 ro,nodev,relatime master:16 - squashfs /dev/loop0 ro
-1068 1060 2:1 / /var/lib/snapd/hostfs/snap/core18/1 ro,nodev,relatime master:17 - squashfs /dev/loop1 ro
-1069 1060 2:2 / /var/lib/snapd/hostfs/snap/test-snapd-mountinfo-classic/1 ro,nodev,relatime master:18 - squashfs /dev/loop2 ro
-1070 1060 2:3 / /var/lib/snapd/hostfs/snap/test-snapd-mountinfo-core16/1 ro,nodev,relatime master:19 - squashfs /dev/loop3 ro
-1071 1060 2:4 / /var/lib/snapd/hostfs/snap/test-snapd-mountinfo-core18/1 ro,nodev,relatime master:20 - squashfs /dev/loop4 ro
-1072 1001 0:0 /var/log /var/log rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
-1073 1001 0:0 /var/snap /var/snap rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
-1074 1001 0:0 /var/tmp /var/tmp rw,relatime master:1 - ext4 /dev/sda1 rw,data=ordered
+1057 1001 1:34 / /usr/share rw,relatime - tmpfs tmpfs rw,mode=755
+1058 1057 1:35 / /usr/share/X11/test rw,relatime - tmpfs tmpfs rw
+1059 1057 2:1 /usr/share/adduser /usr/share/adduser ro,nodev,relatime master:17 - squashfs /dev/loop1 ro
```

Comment:
why is it that with a snap that has core18 base there are all of these additional paths bind mounted from the mimic where as with a core base there is only the first 2? Is that because the mimic bug only is exhibited on snaps using the core18 base snap and not on core base snap? I didn't get that from the PR description or the bug description which is why I'm confused to see all these additional bind mounts

Reply:
@anonymouse64 I'll double check but the contents of `/usr/share` is replicated as distinct mount points. It is possible that `core16` and `core18` simply differ in this regard.

Note that I will change this branch to drop the `mount_id` and `parent_id` fields to make the diffs much more readable. Then we can properly evaluate the behaviour of mimics.

Reply:
I've opened https://github.com/snapcore/snapd/pull/7448 to this effect. I'll merge it into this branch as soon as I have confidence that it can be merged.

Reply:
I've merged 7448 here as I described. As branches land I will take action to reduce the diff in this branch to its essence. As a separate comment, the question raised by @anonymouse64 is valid and stands.

Reply:
I think this mystery is now resolved.